### PR TITLE
Move UserService to Jellyfin.Api

### PR DESCRIPTION
**Pull request by Ullmie02**
Part of #2872

### CODE REVIEW

**Comment by Ullmie02** [comment]
Blocked because of missing authorization policy from #3357

File: Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -215,6 +215,19 @@ private static void AddSwaggerTypeMappings(this SwaggerGenOptions options)
...
+                {
+                    Type = "object",
+                    Properties = typeof(ImageType).GetEnumNames().ToDictionary(
+                        name => name,
+                        name => new OpenApiSchema
+                        {
+                            Type = "string",
+                            Format = "string"
+                        })
+                });
```
Comment (crobibero) [inline Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs:230]:
```suggestion
            /*
             * Support BlurHash dictionary
             */
            options.MapType<Dictionary<ImageType, Dictionary<string, string>>>(() =>
                new OpenApiSchema
                {
                    Type = "object",
                    Properties = typeof(ImageType).GetEnumNames().ToDictionary(
                        name => name,
                        name => new OpenApiSchema
                        {
                            Type = "object", Properties = new Dictionary<string, OpenApiSchema>
                            {
                                {
                                    "string",
                                    new OpenApiSchema
                                    {
                                        Type = "string",
                                        Format = "string"
                                    }
                                }
                            }
                        })
                });
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+        /// <response code="200">Public users returned.</response>
+        /// <returns>An <see cref="IEnumerable{UserDto}"/> containing the public users.</returns>
+        [HttpGet("Public")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<IEnumerable<UserDto>> GetPublicUsers()
+        {
+            // If the startup wizard hasn't been completed then just return all users
+            if (!_config.Configuration.IsStartupWizardCompleted)
+            {
+                return Ok(GetUsers(false, false, false).Value);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:100]:
```suggestion
                return Ok(Get(false, false, false, false));
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+                if (!_networkManager.IsInLocalNetwork(HttpContext.Connection.RemoteIpAddress.ToString()))
+                {
+                    users = users.Where(i => i.HasPermission(PermissionKind.EnableRemoteAccess));
+                }
+            }
+
+            var result = users
+                .OrderBy(u => u.Username)
+                .Select(i => _userManager.GetUserDto(i, HttpContext.Connection.RemoteIpAddress.ToString()))
+                .ToArray();
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:549]:
```suggestion
                .Select(i => _userManager.GetUserDto(i, HttpContext.Connection.RemoteIpAddress.ToString()));
```

Don't need `.ToArray()` if returning `IEnumerable`

File: Jellyfin.Api/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+        {
+            var user = _userManager.GetUserById(id);
+
+            if (user == null)
+            {
+                return NotFound("User not found");
+            }
+
+            var result = _userManager.GetUserDto(user, HttpContext.Connection.RemoteIpAddress.ToString());
+            return Ok(result);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:127]:
```suggestion
            return result;
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+                    AppVersion = auth.Version,
+                    DeviceId = auth.DeviceId,
+                    DeviceName = auth.Device,
+                    Password = request.Pw,
+                    PasswordSha1 = request.Password,
+                    RemoteEndPoint = HttpContext.Connection.RemoteIpAddress.ToString(),
+                    Username = request.Username
+                }).ConfigureAwait(false);
+
+                return Ok(result);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:222]:
```suggestion
                return result;
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+
+            // no need to authenticate password for new user
+            if (request.Password != null)
+            {
+                await _userManager.ChangePassword(newUser, request.Password).ConfigureAwait(false);
+            }
+
+            var result = _userManager.GetUserDto(newUser, HttpContext.Connection.RemoteIpAddress.ToString());
+
+            return Ok(result);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:479]:
```suggestion
            return result;
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+        [HttpPost("ForgotPassword")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<ForgotPasswordResult>> ForgotPassword([FromBody] string enteredUsername)
+        {
+            var isLocal = HttpContext.Connection.RemoteIpAddress.Equals(HttpContext.Connection.LocalIpAddress)
+                          || _networkManager.IsInLocalNetwork(HttpContext.Connection.RemoteIpAddress.ToString());
+
+            var result = await _userManager.StartForgotPasswordProcess(enteredUsername, isLocal).ConfigureAwait(false);
+
+            return Ok(result);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:497]:
```suggestion
            return result;
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+        /// </summary>
+        /// <param name="pin">The pin.</param>
+        /// <response code="200">Pin reset process started.</response>
+        /// <returns>A <see cref="Task"/> containing a <see cref="PinRedeemResult"/>.</returns>
+        [HttpPost("ForgotPassword/Pin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<PinRedeemResult>> ForgotPasswordPin([FromBody] string pin)
+        {
+            var result = await _userManager.RedeemPasswordResetPin(pin).ConfigureAwait(false);
+            return Ok(result);
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:511]:
```suggestion
            return result;
```

File: Jellyfin.Api/Controllers/UserController.cs [hunk 9]
```diff
@@ -0,0 +1,554 @@
+﻿#nullable enable
+#pragma warning disable CA1801
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/UserController.cs:2]:
```suggestion
```

Don't need `#nullable enable` (just informed)
Remove pragma in favor of `SuppressMessage` (see https://github.com/jellyfin/jellyfin/issues/2872#issuecomment-617845103 for usage)

**Pull request merged**